Snakefile

```diff
@@ -663,3 +672,59 @@ rule post_process:
         ts = tskit.load(input[0])
         ts = tsinfer.post_process(ts)
         ts.dump(output[0])
+
+
+rule split_disjoint_nodes:
+    input:
+        data_dir
+        / "trees"
+        / "{subset_name}-{region_name}-{filter_set}"
+        / "{subset_name}-{region_name}-{filter_set}-truncate-{lower}-{upper}-{multiplier}-mm{mismatch}-post-processed.trees",
+    output:
+        data_dir
+        / "trees"
+        / "{subset_name}-{region_name}-{filter_set}"
+        / "{subset_name}-{region_name}-{filter_set}-truncate-{lower}-{upper}-{multiplier}-mm{mismatch}-post-processed-SDN.trees",
+    threads: 1
+    resources:
+        mem_mb=64000,
+        time_min=4 * 60,
+        runtime=4 * 60,
+    run:
+        import tskit
+        import tsdate
+
+        ts = tskit.load(input[0])
```

Comment:
You need to simplify the trees before dating: I don't think it matters whether you do it before or after splitting disjoint nodes. I don't think simplification should be a separate rule, because ultimately both simplification and SDN will be part of a single preprocessing step within tsdate.

Reply:
I thought simplification was part of `tsinfer.post_process`? https://github.com/tskit-dev/tsinfer/blob/92b3a3734dfec53a2a0d74fe2071c13d24c8a931/tsinfer/inference.py#L2674

Reply:
`tsinfer` doesn't fully simplify. You also need to remove unary nodes.

I do this using `tsdate.pre_process` (which removes data poor regions too, but I guess you've done that elsewhere)

I think it might matter which order you do the split_disjoint_nodes. Best to do so after full simplification.

Reply:
Right, gotcha

Reply:
Ok fellas, check it out now.